### DISCUSSION TITLE
Email invites not working

### DISCUSSION THREAD

**Discussion by karianmash** [docmost/docmost#472] [Help / Questions]
I uses Docmost everyday and I honestly love it. However, I have never been able to configure SMTP, any assistant will be appreciated.

**Comment by Philipinho** [comment]
Are you using any SMTP provider?

**Reply by metrbox** [reply]
Hi buddy, I have similar issue. I use mailgun and also tried postmark.
Thanks

**Reply by adrian-dumitrescu-tfs** [reply]
How do you hang mailgun into docmost?

**Reply by adrian-dumitrescu-tfs** [reply]
I'm getting "Error processing send-email job. Reason: self-signed certificate" while using unencrypted SMTP.

**Comment by Philipinho** [comment]
@karianmash and @metrbox, what errors do you get when you check the server logs?

**Comment by polargh** [comment]
hi, currently experiencing something similar - my emails are not sending, it's not even giving requests to postmark or smtp

**Comment by WavedirectTel** [comment]
Would be nice if we could install this on something other than docker.

**Reply by Philipinho** [reply]
You can clone and build it yourself without docker.

```bash
git clone https://github.com/docmost/docmost
cd docmost

npm install -g pnpm

pnpm install
pnpm build
```
After a successful build, start the server with:
```bash
pnpm start
```

**Reply by WavedirectTel** [reply]
> You can clone and build it yourself without docker.
> 
> ```shell
> git clone https://github.com/docmost/docmost
> cd docmost
> 
> npm install -g pnpm
> 
> pnpm install
> pnpm build
> ```
> 
> After a successful build, start the server with:
> 
> ```shell
> pnpm start
> ```

Thank you for this. I gave it a shot on Fedora and it works but it was a bit of a grind.  I also had to set up pgsql among other things.  Any chance you could write up a full installation doc for Ubuntu/Fedora ? All steps needed,  the minimum requirements etc?  How to update going forward etc?  

This will help some of us who are not comfortable using docker in prod.  

Also how we'd set up the email.  How does it send invites natively?  (I know I'll need the same .env parameters) but how does it send?  Sendmail mailx postfix etc?  (EDIT) It just worked out of the box.

**Comment by WavedirectTel** [comment]
I have been unable to get email to send for invites as well, and I have the correct values in the variables.  Having some form of "Test" button integrated into the software itself in the "settings" section would be ideal for future development.  Most programs would have something similar to click test and it would confirm a successful delivery to the server or at least spit an error.  

Right now there is no error we can use to troubleshoot.

**Reply by Philipinho** [reply]
Try resending the invite mail, and then check the server logs for any error.

**Reply by WavedirectTel** [reply]
Not a fan of docker, a bit old school,  but yes I'll and check it out thanks.  This software is freaking awesome.

**Comment by bwnstck** [comment]
Got the same issue on a self hosted Coolify Instance with Docmost.
Locally there is no Problem and all mail services / configurations work as expected or throw an error in the Redis Server if something is misconfigured.

But on my production Instance I can't get any email service working. Neither different smtp servers nor different configurations with or without secure connection.
At the same time I can't get any log from the Redis Server, that any error occurred.

Is it maybe a problem on the connection layer? Docker/Coolify/Server Settings? I turned off my firewall for an hour to check if any port gets blocked, but again no success :)

Iam happy about any ideas <3

**Reply by Philipinho** [reply]
The logs should be in the main Docmost container.
Looking forward to your feedback.

**Reply by WavedirectTel** [reply]
I actually couldn't find the logs inside any container nor was it in the main docker logs for email, but I discovered my issue was that you need to put the environment variables in the docker compose file for the stack,  not the individual container.  After I did that it worked.  I'm using portainer.

**Comment by Grandtheft430** [comment]
Hello, @Philipinho!

First of all - thank you for your work, Docmost is an amazing project and I love using it so far.
But I've got some problems with the emails as well.

Long story short, I've tried to use GMail and Yandex as SMTP providers. With some trial and error I've found a combination of parameters which seem to work but whenever i try to send an invite, i get this message in the logs:

`Error processing send-email job. Reason: 38024A604C700000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:354:`

However, if i try to test the exact same parametres in other service (like gmass) - it works like a charm. Could you look into it, please?
Ready to provide as much info as i can!

**Reply by Philipinho** [reply]
Can I know what port you are connecting to?

**Reply by Grandtheft430** [reply]
I've tried both 465 and 587 with different combination of SMTP_SECURE as "true" and "false".
I'll update with more info on exact port and SMTP_SECURE values with different SMTP providers later today.

**Reply by Grandtheft430** [reply]
I'm sorry I'm late, but I've tried more settings combinations for Gmail and Yandex but none of them worked for me. 
I couldn't make SMTP2GO account, so i made one on smtp.bz

Works with no problems.

**Comment by BeanHubbleday** [comment]
@Philipinho - A suggestion, as this tripped me up when I first spun up DocMost, in the Docker Compose script included on here, the SMTP info included so the user's who are new to it know to enter some info in there?

**Reply by Philipinho** [reply]
That makes sense.

**Comment by ccigas** [comment]
Im using Mailgun and having issues with Docmost only. My other services are able to send emails without a problem. Also, I dont see any logs that Docmost is even trying to send an email. 

This would not be much of a problem but theres also no invite link. When I click on the menu on to resend the email, I also clicked copy invite link and nothing copies.

Any ideas for what I can try?

```
      MAIL_DRIVER: smtp
      MAIL_FROM_ADDRESS: postmaster@mail.domain.cc
      MAIL_FROM_NAME: Docmost
      SMTP_HOST: smtp.mailgun.org
      SMTP_PORT: 587
      SMTP_USERNAME: postmaster@mail.domain.cc
      SMTP_PASSWORD: <password>
      SMTP_SECURE: true
      SMTP_IGNORETLS: false
```

**Reply by Philipinho** [reply]
Check the docker container logs. There should be an error message.
```sh
docker logs container_id_here
```

**Reply by ccigas** [reply]
Well there was no logs, thats because I put my vars on the postgress container instead of docmost. Fixed that and all good now. Sorry to bother.

**Comment by ChingOK** [comment]
I also have this problem，and  my email use self-signed certificate.
"      MAIL_DRIVER: 'smtp'
      MAIL_FROM_ADDRESS: 'oa@****wall.com'
      MAIL_FROM_NAME: 'Docmost'
      SMTP_HOST: 'mail.****wall.com'
      SMTP_PORT: '465'
      SMTP_USERNAME: 'oa@****wall.com'
      SMTP_PASSWORD: 'P0009r4d'
      SMTP_SECURE: 'true'
"
 And I add the config:      SMTP_IGNORETLS: 'true'     NODE_TLS_REJECT_UNAUTHORIZED: '0'   ,it still can't work .  I add the certs to the docker， it still can't work .
error:"docmost-docmost-1  | [Nest] 41  - 03/21/2025, 5:17:03 PM   ERROR [EmailProcessor] Error processing senF0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/rec".

And,If    SMTP_PORT: '25'  SMTP_SECURE: 'false'。it still can‘t work.
error: docmost  | [Nest] 41  - 03/28/2025, 2:22:34 AM   ERROR [EmailProcessor] Error processing send-email job. Reason: Invalid login: 535 5.7.3 Authentication unsuccessful

I used the docmost:v0.9.0

**Reply by wangjc683** [reply]
same problem here as well, 

ERROR [EmailProcessor] Error processing send-email job. Reason: 58729988A17F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:354:

Using smtp.gmail.com and 465 port with SMTP_SECURE: true. It's working in my other docker container app.
